### PR TITLE
New message template creation forms

### PR DESCRIPTION
**Pull request by Thurfjell**
### Ticket:

- [17100](https://dev.azure.com/OGCIO-Digital-Services/Digital%20Services%20Programme/_workitems/edit/17100)

### Description

Updated template creations with two forms for languages to be filled, then a preview and finally create/update.

## Type

- [ ] **Dependency upgrade**
- [ ] **Bug fix**
- [ ] **New feature**
- [ ] **Dev change**

### Checklist:

- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] This change might impact the developer experience of others and should be communicated

### Screenshots:

N/A

### CODE REVIEW

File: apps/messaging/app/[locale]/admin/message-templates/page.tsx [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+  }
+
+  let messageNameToDelete: string | undefined;
+
+  if (props.searchParams?.delete_id) {
+    const { userId: authToken } = await PgSessions.get();
+    const client = new Messaging(authToken);
+    const tmpl = await client.getTemplate(props.searchParams?.delete_id);
+    const content =
+      tmpl.data?.contents.find((content) => content.lang === "en") ||
```
Comment (SamSalvatico) [inline apps/messaging/app/[locale]/admin/message-templates/page.tsx:38]: please set `en` as const somewhere so we can reuse it

File: apps/messaging/app/[locale]/admin/message-templates/page.tsx [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+      value: "en",
+    });
+    redirect(url.href);
+  }
+
+  let messageNameToDelete: string | undefined;
+
+  if (props.searchParams?.delete_id) {
+    const { userId: authToken } = await PgSessions.get();
+    const client = new Messaging(authToken);
```
Comment (SamSalvatico) [inline apps/messaging/app/[locale]/admin/message-templates/page.tsx:35]: I think we could initialize the client once to reuse it

File: apps/messaging/app/utils/routes.ts [hunk 3]
```diff
@@ -20,3 +20,22 @@ export const providerRoutes = {
...
+export const templateRoutes = {
+  slug: "template",
+  url: `${messageTemplates.url}/template`,
+};
+
+export function urlWithSearchParams(
+  dir: string,
+  ...searchParams: { key: string; value?: string }[]
+): URL {
+  console.log(dir);
```
Comment (SamSalvatico) [inline apps/messaging/app/utils/routes.ts:33]: intruder

File: apps/messaging/messages/en.json [hunk 4]
```diff
@@ -47,7 +47,10 @@
     "saveButton": "Save",
     "updateTemplateButton": "Update template",
     "createTemplateButton": "Create template",
-    "backLink": "Back"
+    "backLink": "Back",
+    "menuEnglish": "English",
+    "menuGaelic": "Gaelic",
```
Comment (SamSalvatico) [inline apps/messaging/messages/en.json:52]: I think we should use `Gaeilge` as Francesco did
https://github.com/ogcio/life-events/pull/350/files

File: apps/messaging/app/utils/routes.ts (hunk 3 above)
Comment (Thurfjell) [reply apps/messaging/app/utils/routes.ts:33]: 👿

Comment (Thurfjell) [inline apps/messaging/app/utils/routes.ts:33]:
```suggestion
```

**Review by SamSalvatico — APPROVED** [review]

**Pull request merged**